app/controllers/discovered_hosts_controller.rb

```diff
@@ -171,6 +174,29 @@ def auto_provision_all
 
   private
 
+  def init_regex_and_categories
+    hightlights = Setting[:discovery_facts_highlights].empty? ? /^(productname|memorysize|interface|manufacturer|architecture|macaddress$|processorcount|physicalprocessorcount|discovery_boot|ipaddress$)/ : Regexp.new(eval(Setting[:discovery_facts_highlights]))
+    storage = Setting[:discovery_facts_storage].empty? ? /^blockdevice/ : Regexp.new(eval(Setting[:discovery_facts_storage]))
+    hardware = Setting[:discovery_facts_hardware].empty? ? /^(hardw|manufacturer|memo|process)/ : Regexp.new(eval(Setting[:discovery_facts__hardware]))
+    network = Setting[:discovery_facts_network].empty? ? /^(ipaddress|interfaces|dhcp|fqdn|hostname|link|mtu|net|macaddress|wol|port|speed)/ : Regexp.new(eval(Setting[:discovery_facts_network]))
+    software = Setting[:discovery_facts_software].empty? ? /^(bios|os|discovery)/ : Regexp.new(eval(Setting[:discovery_facts_software]))
+    ipmi = Setting[:discovery_facts_ipmi].empty? ? /^ipmi/ : Regexp.new(eval(Setting[:discovery_facts_ipmi]))
+    @regex_array = [hightlights, storage, hardware, network, software, ipmi, "misceleneous"]
```

Comment:
Use symbol instead. Alternatively, you can use the index, but I kinda like this solution. Another more clever approach is to create a class named `AlwaysMatcher` with one method `def match; true; end` adding it as the last element. Then the loop can be even more simple (it will always match). But this works for me too.